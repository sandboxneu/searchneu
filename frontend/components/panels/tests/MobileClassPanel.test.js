/*
 * This file is part of Search NEU and licensed under AGPL3.
 * See the license file in the root folder for details.
 */

import React from 'react';
import MockDate from 'mockdate';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import mockData from './mockData';
import MobileClassPanel from '../MobileClassPanel';

Enzyme.configure({ adapter: new Adapter() });

beforeAll(() => {
  MockDate.set('Mon Nov 26 2017 00:00:00 -0000');
});

afterAll(() => {
  MockDate.reset();
});

it('should render some stuff', () => {
  const wrapper = shallow(<MobileClassPanel aClass={ mockData.cs1210 } history={ null } />);
  expect(wrapper).toMatchSnapshot();
});
