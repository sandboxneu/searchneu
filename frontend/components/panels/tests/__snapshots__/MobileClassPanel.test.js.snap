// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render some stuff 1`] = `
<div
  className="class-panel-container ui segment mobile"
>
  <div
    className="header"
    onClick={[Function]}
    role="button"
    tabIndex={0}
  >
    <img
      alt=""
      className="chevron"
      src="file"
    />
    <span
      className="classTitle"
    >
      CS
       
      1210
      : 
      Computer Science/Information Science Overview 2: Co-op Preparation
    </span>
  </div>
  <span
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="panel-body"
    >
      <span>
        <span>
          Continues the preparation of students for careers in the computing and...
              
        </span>
        <div>
          <a
            onClick={[Function]}
            role="button"
            style={
              Object {
                "display": "inline-block",
              }
            }
            tabIndex={0}
          >
            Show More...
          </a>
        </div>
      </span>
    </div>
    <MobileSectionPanel
      key="neu.edu/201810/CS/1210/13502"
      section={
        Section {
          "classId": "1210",
          "crn": "13502",
          "dataStatus": undefined,
          "honors": undefined,
          "host": "neu.edu",
          "lastUpdateTime": 1511131914029,
          "meetings": Array [
            Meeting {
              "endDate": "2017-12-07T00:00:00.000Z",
              "location": "Snell Engineering Center 108",
              "startDate": "2017-09-07T00:00:00.000Z",
              "times": Array [
                Object {
                  "end": "1970-01-06T11:30:00.000Z",
                  "start": "1970-01-06T09:50:00.000Z",
                },
              ],
            },
          ],
          "online": false,
          "profs": Array [
            "Jennifer Anne Shire",
          ],
          "seatsCapacity": 19,
          "seatsRemaining": 0,
          "subject": "CS",
          "termId": "201810",
          "url": "https://wl11gp.neu.edu/udcprod8/bwckschd.p_disp_detail_sched?term_in=201810&crn_in=13502",
          "waitCapacity": 0,
          "waitRemaining": 0,
        }
      }
      showNotificationSwitches={false}
    />
    <div
      className="more-sections-button"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    >
      Show More...
    </div>
  </span>
</div>
`;
